sdk/src/main/java/com/mapbox/maps/renderer/widget/WidgetPosition.kt

```diff
@@ -1,17 +1,135 @@
 package com.mapbox.maps.renderer.widget
 
 import com.mapbox.maps.MapboxExperimental
+import java.util.Objects
 
 /**
- * Specifies widget position relative to the screen.
- * @param horizontal position relative to the screen.
- * @param vertical position relative to screen.
+ * Specifies widget position relative to the map.
+ *
+ * @property horizontalAlignment the horizontal position relative to the map.
+ * @property verticalAlignment the vertical position relative to the map.
+ * @property offsetX the horizontal offset in pixels towards the right of the map.
+ * @property offsetY the vertical offset in pixels towards the bottom of the map.
  */
 @MapboxExperimental
-class WidgetPosition(
-  val horizontal: Horizontal,
-  val vertical: Vertical,
+class WidgetPosition private constructor(
+  val horizontalAlignment: Horizontal,
+  val verticalAlignment: Vertical,
+  val offsetX: Float,
+  val offsetY: Float
 ) {
+  /**
+   * Deprecated constructor for WidgetPosition.
+   *
+   * @param horizontal the horizontal position relative to the map.
+   * @param vertical the vertical position relative to the map.
+   */
+  @Deprecated(
+    message = "Direct constructor of WidgetPosition is deprecated, and might be removed in future releases.",
+    replaceWith = ReplaceWith("WidgetPosition.Builder()")
+  )
+  constructor(horizontal: Horizontal, vertical: Vertical) : this(horizontal, vertical, 0f, 0f)
+
+  /**
+   * The horizontal position relative to the map.
+   */
+  @Deprecated(
+    message = "horizontal has been renamed to horizontalAlignment, and might be removed in future releases.",
+    replaceWith = ReplaceWith("horizontalAlignment")
+  )
+  val horizontal = horizontalAlignment
+
+  /**
+   * The vertical position relative to the map.
+   */
+  @Deprecated(
+    message = "vertical has been renamed to verticalAlignment, and might be removed in future releases.",
+    replaceWith = ReplaceWith("verticalAlignment")
+  )
+  val vertical = verticalAlignment
+
+  /**
+   * Returns a String for the object.
+   */
+  override fun toString() =
+    "WidgetPosition(horizontalAlignment=$horizontalAlignment,verticalAlignment=$verticalAlignment,offsetX=$offsetX,offsetY=$offsetY)"
+
+  /**
+   * Indicates whether some other object is "equal to" this one.
+   */
+  override fun equals(other: Any?) = other is WidgetPosition &&
+    horizontalAlignment == other.horizontalAlignment &&
+    verticalAlignment == other.verticalAlignment &&
+    offsetX.compareTo(other.offsetX) == 0 &&
+    offsetY.compareTo(other.offsetY) == 0
```

Comment:
Nit:
```suggestion
    offsetX == other.offsetX &&
    offsetY == other.offsetY
```

Reply:
The above is suggested by [EqualsVerifier](https://jqno.nl/equalsverifier/errormessages/float-equals-doesnt-use-floatcompare-for-field-foo/) 
> For more information, go to: https://www.jqno.nl/equalsverifier/errormessages
java.lang.AssertionError: EqualsVerifier found a problem in class com.mapbox.maps.renderer.widget.WidgetPosition.
-> Float: equals doesn't use Float.compare for field offsetX.

>  You should never use == to compare doubles or floats in an equals method. Always use [Double.compare](http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#compare(double,%20double%29)) or [Double.compareTo](http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html#compareTo(java.lang.Double%29)) instead.
> Josh Bloch explains this in his book Effective Java. The short summary is that this method will do the right thing when confronted with NaN and positive and negative infinities. For example, Float.NaN is not equal to itself, but it has to be for equals, or you would never be able to retrieve it from a HashMap.

Reply:
Thanks! TIL, I always relied on the automatic one generated by the IDE. I wonder why that one is not using the `compare` 🤔